src/mlpack/bindings/java/CMakeLists.txt

```diff
@@ -0,0 +1,227 @@
+  not_found_return("Not building Java bindings.")
+endif ()
+
+set(JAVA_VERSION 1.7)
+find_package(Java ${JAVA_VERSION} REQUIRED COMPONENTS Development)
+if (Java_FOUND)
+  include(UseJava)
+else()
+  not_found_return("Java components weren't found")
+endif()
+
+find_package(JNI REQUIRED)
+if (NOT JNI_FOUND)
+  not_found_return("JNI wasn't found")
+endif()
+
+set(JNI_INCLUDE ${JNI_INCLUDE_DIRS} CACHE INTERNAL "")
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  find_program(MAVEN_BIN mvn.bat REQUIRED)
+else()
+  find_program(MAVEN_BIN mvn REQUIRED)
+endif()
+
+if (MAVEN_BIN)
+  message(STATUS "Found maven: ${MAVEN}")
+else()
+  not_found_return("Maven not found")
+endif()
+
+set(MAVEN ${MAVEN_BIN} CACHE INTERNAL "")
+set(JAVA ${Java_JAVA_EXECUTABLE} CACHE INTERNAL "")
+
+add_custom_target(java_configure ALL)
+add_custom_target(java ALL DEPENDS java_configure)
+
+# Create directory structure for maven project
+set(MAVEN_BUILD_DIR ${CMAKE_BINARY_DIR}/src/mlpack/bindings/java/maven CACHE INTERNAL "")
+add_custom_command(TARGET java_configure PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${MAVEN_BUILD_DIR}/src/main/java/org/mlpack)
+
+# Create directory to output native libraries to
+set(NATIVE_OUT_DIR ${MAVEN_BUILD_DIR}/native CACHE INTERNAL "")
+add_custom_command(TARGET java_configure PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${NATIVE_OUT_DIR})
+
+# Create directory to generate JNI files to
+set(JNI_GEN_DIR ${CMAKE_BINARY_DIR}/src/mlpack/bindings/java/build/gen CACHE INTERNAL "")
+add_custom_command(TARGET java_configure PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${JNI_GEN_DIR})
+
+set(BUILD_SRC_DIR ${CMAKE_BINARY_DIR}/src/mlpack/bindings/java/build/src CACHE INTERNAL "")
+add_custom_command(TARGET java_configure PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${BUILD_SRC_DIR})
+
+# Copy maven pom.xml
+set(JAVACPP_VERSION 1.5.1)
+
+file(READ "${CMAKE_SOURCE_DIR}/src/mlpack/core/util/version.hpp"
+    VERSION_HPP_CONTENTS)
+string(REGEX REPLACE ".*#define MLPACK_VERSION_MAJOR ([0-9]+).*" "\\1"
+    MLPACK_VERSION_MAJOR "${VERSION_HPP_CONTENTS}")
+string(REGEX REPLACE ".*#define MLPACK_VERSION_MINOR ([0-9]+).*" "\\1"
+    MLPACK_VERSION_MINOR "${VERSION_HPP_CONTENTS}")
+string(REGEX REPLACE ".*#define MLPACK_VERSION_PATCH [\"]?([0-9x]+)[\"]?.*"
+    "\\1" MLPACK_VERSION_PATCH "${VERSION_HPP_CONTENTS}")
+set(PACKAGE_VERSION
+      "${MLPACK_VERSION_MAJOR}.${MLPACK_VERSION_MINOR}.${MLPACK_VERSION_PATCH}")
+
+configure_file(maven/pom.xml.in ${MAVEN_BUILD_DIR}/pom.xml @ONLY)
+
+# Make maven resolve dependencies into lib directory (see pom.xml)
+add_custom_command(TARGET java_configure PRE_BUILD
+    COMMAND ${MAVEN} dependency:copy-dependencies
+    WORKING_DIRECTORY ${MAVEN_BUILD_DIR})
+
+# Javacpp was pulled from maven
+set(JAVACPP ${MAVEN_BUILD_DIR}/lib/javacpp-${JAVACPP_VERSION}.jar CACHE INTERNAL "")
+
+# Compile cli_util library for java
+add_custom_target(jniCLI_configure ALL
```

Comment:
I guess so, the reason for all these issues is a Java convention to name its classes in CamelCase (see comment below). It will be fixed in a new build process I'm working on

Reply:
Fixed in 81ab8ee